src/System.IO.Compression.Brotli/README.md

```diff
@@ -0,0 +1,219 @@
+﻿# System.IO.Compression.Brotli
+
+## Introduction
+
+Brotli is a generic-purpose lossless compression algorithm that compresses data
+using a combination of a modern variant of the LZ77 algorithm, Huffman coding
+and 2nd order context modeling, with a compression ratio comparable to the best
+currently available general-purpose compression methods. It is similar in speed
+with deflate but offers more dense compression.
+
+The specification of the Brotli Compressed Data Format is defined in [RFC 7932](https://www.ietf.org/rfc/rfc7932.txt).
+
+Brotli encoding is supported by most web browsers, major web servers, and some CDNs.
+
+## Brotli Class
+```C#
+public static class Brotli {
+        public static TransformationStatus Compress(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten, ref Brotli.State state);
+        public static TransformationStatus Decompress(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten, ref Brotli.State state);
+        public static TransformationStatus FlushEncoder(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten, ref Brotli.State state, bool isFinished=true);
+        public static int GetMaximumCompressedSize(int inputSize);
+        public struct State : IDisposable {
+            public bool CompressMode { get; }
+            public void Dispose();
+            public void SetQuality();
+            public void SetQuality(uint quality);
+            public void SetWindow();
+            public void SetWindow(uint window);
+        }
+    }
+```
+```out bytesConsumed``` - number of bytes from source, which used in this call
+
+```out bytesWritten``` - number of written to destination bytes 
+
+The ```Compress``` performs data compression. It sets ```source``` data should be compressed and return compressed data in ```destination```. ```state.CompressMode``` should be ```true``` .
+
+
+The ```FlushEncoder``` return compressed data, which have sent to ```state``` using Compress. 
+
+The ```Decompress``` Decompresses the data in ```source``` into ```destination```. ```state.CompressMode``` should be ```false``` .
+
+```State``` uses in both compress and decompress mode to save a temporary status and data of process.
```

Comment:
uses => used.

Also, state should be moved up and described as an opaque state structured that cannot be re-used once you start compressing / decompression data with it.